### PR TITLE
Adding Client hostname to Sql server connection

### LINKED ISSUE #12627: Client hostname not provided when connecting with SqlClient
Connecting to a SQL Server instance with SSMS or with the full .NET framework SqlClient library will generally include the hostname of the client. This is visible on the database when running `EXEC sp_who2`. However, in the current stable build of dotnet, the SqlClient does not provide a hostname for the client when connecting.

**Scenario:** Connecting to a SQL Server instance using SqlClient library in System.Data.SqlClient and observing the connected instance on the SQL Server instance via the sp_who2 stored proc.

```
SqlConnection sql = new SqlConnection(connectionString);
sql.Open();
```

**Expected Behavior:** When using `EXEC sp_who2`, the connection opened in the scenario above would have the hostname of the client populated in the Hostname column.

**Actual Behavior:** The connection created in the scenario is shown, and will have an application name present if provided, but the hostname field will be blank.

This feature makes it much easier to determine who/where connections are for a given sql server instance.

### PR DESCRIPTION
**Pull request by saurabh500**
While connecting to the Sql Server, the SqlClient was sending an empty hostname to the server. 
In this pull request we use using the Environment.MachineName to populate the hostname while connecting to SqlServer.

Fixes #12627 

cc @stephentoub @corivera

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs [hunk 1]
```diff
@@ -521,7 +521,7 @@ internal string ObtainWorkstationId()
                 // permission to obtain Environment.MachineName is Asserted
                 // since permission to open the connection has been granted
                 // the information is shared with the server, but not directly with the user
-                result = string.Empty;
+                result = ADP.MachineName();
```
Comment (justinvp) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs]: Just curious: In .NET Framework, the length of the machine name is validated. Is the length check not needed for .NET Core?

https://github.com/Microsoft/referencesource/blob/master/System.Data/System/Data/SqlClient/SqlConnectionString.cs#L701

**Comment by saurabh500** [comment]
> Is there a test we should add?

@stephentoub  Yes. I will be adding another PR with the test.

**Comment by saurabh500** [comment]
@stephentoub The build for Innerloop Linux ARM Emulator SoftFP Debug Cross...  is failing. What is the full title of this step? I want the bot to execute this step.

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop Linux ARM Emulator SoftFP Debug Cross Build and Test please

**Comment by saurabh500** [comment]
@stephentoub I have added the tests to this PR itself.

I have modified the ManualTest to build on 1.7 to access `Environment.MachineName` The other manual tests succeed with the migration.

**Pull request merged**